src/core/main.cpp

```diff
@@ -298,6 +298,57 @@ void check_debug()
     }
 }
 
+/**
+ * Read the CPU's Hz (based on /proc/cpuinfo Mhz report)
+ * Provide the std::max and std::min values, which might be the case if core are running at power
+ *control states Return true on success, false on any failure
+ **/
+static bool get_cpu_hz(double &hz_min, double &hz_max)
+{
+    FILE *f;
+    char buf[256];
+    bool first_run = true;
+
+    f = fopen("/proc/cpuinfo", "r");
+    if (!f) {
+        return false;
+    }
+
+    while (fgets(buf, sizeof(buf), f)) {
+        double mhz = 0;
+        int rc = 0;
+
+#if defined(__ia64__)
+        rc = sscanf(buf, "itc MHz : %lf", &mhz);
+#elif defined(__powerpc__)
+        rc = sscanf(buf, "clock : %lf", &mhz);
+#elif defined(__aarch64__)
+        rc = sscanf(buf, "BogoMIPS : %lf", &mhz);
+        mhz /= 2;
```

Comment:
why mhz is half of BogoMIPS?

Reply:
on one of our bf3 for example:
BogoMIPS	: 660.55

but:
	Max Speed: 2134 MHz
	Current Speed: 2134 MHz
as you can see from "dmidecode -t 4" output

Reply:
we must solve it here...
in case if defined(__aarch64__) is true - "mhz" will be incorrect

Reply:
check if we can count on /sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_max_freq
or maybe
/sys/devices/system/cpu/cpu0/cpufreq/cpuinfo_cur_freq

Reply:
I won't make this code change in this PR.